yt_dlp/extractor/redgifs.py

```diff
@@ -80,15 +75,196 @@ def _real_extract(self, url):
             })
         self._sort_formats(formats)
 
+        webpage_url = f'https://redgifs.com/watch/{video_id}'
+
         return {
             'id': video_id,
-            'title': ' '.join(gif.get('tags') or []) or 'RedGifs',
-            'timestamp': int_or_none(gif.get('createDate')),
-            'uploader': gif.get('userName'),
-            'duration': int_or_none(gif.get('duration')),
-            'view_count': int_or_none(gif.get('views')),
-            'like_count': int_or_none(gif.get('likes')),
-            'categories': gif.get('tags') or [],
+            'title': ' '.join(gif_data.get('tags') or []) or 'RedGifs',
+            'timestamp': int_or_none(gif_data.get('createDate')),
+            'uploader': gif_data.get('userName'),
+            'duration': int_or_none(gif_data.get('duration')),
+            'view_count': int_or_none(gif_data.get('views')),
+            'like_count': int_or_none(gif_data.get('likes')),
+            'categories': gif_data.get('tags') or [],
+            'webpage_url': webpage_url,
+            'tags': gif_data.get('tags'),
             'age_limit': 18,
             'formats': formats,
         }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url).lower()
+
+        video_info = self._download_json(
+            'https://api.redgifs.com/v2/gifs/%s' % video_id,
+            video_id, 'Downloading video info')
+        if 'error' in video_info:
+            raise ExtractorError(f'RedGifs said: {video_info["error"]}', expected=True)
+
+        return self._parse_gif_data(video_info['gif'])
+
+
+class RedGifsSearchIE(RedGifsIE):
```

Comment:
Do not subclass from another user-facing extractor - that's just asking for trouble. Make a `RedGifsBaseInfoExtractor` from which all of these can be subclassed

Reply:
Makes sense, was wondering that. Done